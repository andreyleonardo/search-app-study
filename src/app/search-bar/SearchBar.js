import React from 'react';
import PropTypes from 'prop-types';
import SearchIcon from './icons/SearchIcon';
import './SearchBar.scss';

function onSubmit(event, callback) {
  event.preventDefault();
  if (callback) {
    callback();
  }
}

const SearchBar = ({ onChange, handleSubmit, name }) => (
  <form
    id="search-form-id"
    className="search_bar"
    name="search_bar_form"
    onSubmit={event => onSubmit(event, handleSubmit)}
  >
    <label htmlFor="search-input-id" className="search_bar__label">
      {"Let's find a great experience"}
    </label>
    <input
      name={name}
      autoComplete="off"
      className="search_bar__input"
      id="search-input-id"
      onChange={onChange}
    />
    <button type="submit" className="search_bar__button">
      <SearchIcon />
    </button>
  </form>
);

SearchBar.propTypes = {
  onChange: PropTypes.func.isRequired,
  handleSubmit: PropTypes.func,
  name: PropTypes.string
};

SearchBar.defaultProps = {
  name: 'search_bar_input',
  handleSubmit: null
};

export default SearchBar;
