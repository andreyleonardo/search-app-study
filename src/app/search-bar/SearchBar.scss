.search_bar {
  margin: 10px;
  &__input {
    transition: box-shadow 0.3s;
  }
  &__input:focus {
    box-shadow: 0px 0px 8px 2px rgba(48,189,255,1);
  }
}
