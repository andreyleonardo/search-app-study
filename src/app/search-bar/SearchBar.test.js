import React from 'react';
import SearchBar from './SearchBar';

describe('Search Bar', () => {
  let props;
  let searchBar;
  beforeEach(() => {
    props = {
      onChange: jest.fn(),
      handleSubmit: jest.fn()
    };
    searchBar = shallow(<SearchBar {...props} />);
  });

  it('renders without crashing', () => {
    expect(searchBar).toMatchSnapshot();
  });

  it('calls onChange function when typing', () => {
    searchBar.find('input#search-input-id').simulate('change');

    expect(props.onChange).toHaveBeenCalledTimes(1);
  });

  it('calls handleSubmit function when present and submits form', () => {
    searchBar.find('form#search-form-id').simulate('submit', { preventDefault: jest.fn() });

    expect(props.handleSubmit).toHaveBeenCalledTimes(1);
  });

  it('does not calls handleSubmit function when is not present and submits form', () => {
    searchBar.setProps({ handleSubmit: null });
    searchBar.find('form#search-form-id').simulate('submit', { preventDefault: jest.fn() });

    expect(props.handleSubmit).toHaveBeenCalledTimes(0);
  });
});
