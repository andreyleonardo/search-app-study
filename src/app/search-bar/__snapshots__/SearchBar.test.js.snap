// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Search Bar renders without crashing 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <SearchBar
    handleSubmit={[MockFunction]}
    name="search_bar_input"
    onChange={[MockFunction]}
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <label
          className="search_bar__label"
          htmlFor="search-input-id"
        >
          Let's find a great experience
        </label>,
        <input
          autoComplete="off"
          className="search_bar__input"
          id="search-input-id"
          name="search_bar_input"
          onChange={[MockFunction]}
        />,
        <button
          className="search_bar__button"
          type="submit"
        >
          <SearchIcon
            height="32"
            viewBox="0 0 32 32"
            width="32"
          />
        </button>,
      ],
      "className": "search_bar",
      "id": "search-form-id",
      "name": "search_bar_form",
      "onSubmit": [Function],
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": "Let's find a great experience",
          "className": "search_bar__label",
          "htmlFor": "search-input-id",
        },
        "ref": null,
        "rendered": "Let's find a great experience",
        "type": "label",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "autoComplete": "off",
          "className": "search_bar__input",
          "id": "search-input-id",
          "name": "search_bar_input",
          "onChange": [MockFunction],
        },
        "ref": null,
        "rendered": null,
        "type": "input",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": <SearchIcon
            height="32"
            viewBox="0 0 32 32"
            width="32"
          />,
          "className": "search_bar__button",
          "type": "submit",
        },
        "ref": null,
        "rendered": Object {
          "instance": null,
          "key": undefined,
          "nodeType": "function",
          "props": Object {
            "fill": undefined,
            "height": "32",
            "viewBox": "0 0 32 32",
            "width": "32",
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        "type": "button",
      },
    ],
    "type": "form",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <label
            className="search_bar__label"
            htmlFor="search-input-id"
          >
            Let's find a great experience
          </label>,
          <input
            autoComplete="off"
            className="search_bar__input"
            id="search-input-id"
            name="search_bar_input"
            onChange={[MockFunction]}
          />,
          <button
            className="search_bar__button"
            type="submit"
          >
            <SearchIcon
              height="32"
              viewBox="0 0 32 32"
              width="32"
            />
          </button>,
        ],
        "className": "search_bar",
        "id": "search-form-id",
        "name": "search_bar_form",
        "onSubmit": [Function],
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": "Let's find a great experience",
            "className": "search_bar__label",
            "htmlFor": "search-input-id",
          },
          "ref": null,
          "rendered": "Let's find a great experience",
          "type": "label",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "autoComplete": "off",
            "className": "search_bar__input",
            "id": "search-input-id",
            "name": "search_bar_input",
            "onChange": [MockFunction],
          },
          "ref": null,
          "rendered": null,
          "type": "input",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": <SearchIcon
              height="32"
              viewBox="0 0 32 32"
              width="32"
            />,
            "className": "search_bar__button",
            "type": "submit",
          },
          "ref": null,
          "rendered": Object {
            "instance": null,
            "key": undefined,
            "nodeType": "function",
            "props": Object {
              "fill": undefined,
              "height": "32",
              "viewBox": "0 0 32 32",
              "width": "32",
            },
            "ref": null,
            "rendered": null,
            "type": [Function],
          },
          "type": "button",
        },
      ],
      "type": "form",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getDerivedStateFromProps": true,
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
  },
}
`;
